### PR TITLE
Update gmap Flag  (gmap Version 2016-09-23)

### PR DESCRIPTION
**Pull request by Imoteph**
Changed --intronlength to the flag --max-intronlength-middle= and --max-intronlength-ends

### CODE REVIEW

**Comment by brianjohnhaas** [comment]
Thanks for the pull request!  For backwards compatibility, perhaps we should check which version of gmap is being used and then apply the proper parameter settings.

**Comment by Imoteph** [comment]
Agreed, that would be the best solution! Maybe another test to add would be to see which version of gmap has to be called (gmap or gmapl) as this also can break the pipeline.

File: PerlLib/GFF3_utils.pm [hunk 1]
```diff
@@ -55,7 +55,7 @@ sub index_GFF3_gene_objs {
 
         my @x = split (/\t/);
 
-		unless (scalar @x >= 9) {
+		unless (@x == 9) {
```
Comment (brianjohnhaas) [inline PerlLib/GFF3_utils.pm]: sometimes the gff3 files embed extra columns for metadata.  I'm not sure this change is critical so will leave it be for now.

File: scripts/process_GMAP_alignments_gff3_chimeras_ok.pl [hunk 2]
```diff
@@ -88,7 +88,7 @@
 
 	my $cmd = "gmap -D $genomeBaseDir -d $genomeDir $transcriptDB -f $format -n $num_gmap_top_hits -x 50 -t $CPU -B 5 ";
 	if ($max_intron) {
-		$cmd .= " --intronlength=$max_intron ";
+		 $cmd .= "--max-intronlength-middle=".$max_intron." --max-intronlength-ends=".$max_intron;
```
Comment (brianjohnhaas) [inline scripts/process_GMAP_alignments_gff3_chimeras_ok.pl]: looks like this was captured as:

if ($max_intron) {
                my $gmapv=`gmap --version 2>&1 | head -1`;
                chomp($gmapv);
                $gmapv=~/GMAP\s+version\s+(\d{4})-(\d{2})-.*/;
                if( ($1 == 2016 && $2 >= 5) || $1 > 2016){
                        $cmd .= " --max-intronlength-middle=$max_intron --max-intronlength-ends=$max_intron ";
                }else{
                        $cmd .= " --intronlength=$max_intron ";
                }

**Pull request closed without merging**